flink-libraries/flink-table/src/main/scala/org/apache/flink/api/table/runtime/aggregate/AvgAggregate.scala

```diff
@@ -66,51 +66,63 @@ abstract class IntegralAvgAggregate[T] extends AvgAggregate[T] {
   def doPrepare(value: Any, partial: Row): Unit
 }
 
-class ByteAvgAggregate extends IntegralAvgAggregate[Byte] {
+class ByteAvgAggregate[T] extends IntegralAvgAggregate[T] {
```

Comment:
I think it would be cleaner to keep the `Byte` type parameter.
How about we add an abstract method `def doEvaluate(buffer: Row): Any` to `IntegralAvgAggregate`. 
The subclasses of `IntegralAvgAggregate` implement `doEvaluate` like `ByteAvgAggregate`:

```
override def doEvaluate(buffer: Row): Any = {
  val bufferSum = buffer.productElement(partialSumIndex).asInstanceOf[Long]
  val bufferCount = buffer.productElement(partialCountIndex).asInstanceOf[Long]
  if (bufferCount == 0L) {
    null
  } else {
    (bufferSum / bufferCount).toByte
  }
}
```

and return an `Any` which is casted to `T` by `IntegralAvgAggregate.evaluate()` as follows:


```
override def evaluate(buffer: Row): T = {
  doEvaluate(buffer).asInstanceOf[T]
}
```

Same for the `FloatingAvgAggregate` and its subclasses.